models/schedule_item.go

```diff
@@ -66,11 +68,101 @@ func NewTimeslotItem(t *myradio.Timeslot, finish time.Time, u func(*myradio.Time
 		Desc:    t.Description,
 		Start:   t.StartTime,
 		Finish:  finish,
-		Block:   "regular", // TODO(MattWindsor91): get this from elsewhere
+		Block:   getBock(t.Title, t.StartTime), // TODO(MattWindsor91): get this from elsewhere
 		PageURL: url.Path,
 	}, nil
 }
 
+func getBock(name string, StartTime time.Time) string{
```

Comment:
(This is going to have a lot of references to things that aren't publicly available, for which I apologise to anyone from outside looking in)

This should either be in a configuration file (like it is with 2013site) or retrieved from the MyRadio API; one of the reasons I never put this in was because I was expecting the latter but waiting on someone to design the API for doing so.

From experience, putting it somewhere only compteam and friends can get to it is a world of pain.  I thought that changing a config file every now and then would be easy, but forgot that:

1) there'd be a noticeable delay between committee asking for a show to be 'coloured in' and compteam getting around to doing it;
2) if the config file isn't in VCS, it's hard to copy around to where it needs to be;
3) if the config file *is* in VCS (and ours is), it means every change needs a push to GitHub and all of the code review and bureaucracy that entails;
4) it means that compteam are involved in a decision and process that doesn't even involve compteam---it's really the programme controller's job to decide how the schedule gets divided up.

This is entirely separate from the problem of how we decide which shows are in which blocks, and even then the name-matching approach doesn't scale as well as it seems on first blush.  See the amount of changes that have been needed to 2013-site's config file.